src/main/java/refactoringml/App.java

```diff
@@ -108,173 +118,203 @@ public App (String datasetName,
-				if(thereIsRefactoringToProcess)
-					//remove all not studied refactorings from the list
-					refactoringsToProcess = refactoringsToProcess.stream().filter(RefactoringUtils::isStudied).collect(Collectors.toList());
+				if (thereIsRefactoringToProcess)
+					// remove all not studied refactorings from the list
+					refactoringsToProcess = refactoringsToProcess.stream().filter(RefactoringUtils::isStudied)
+							.collect(Collectors.toList());
 
-				//check if refactoring miner detected a refactoring we study
+				// check if refactoring miner detected a refactoring we study
 				if (thereIsRefactoringToProcess && !refactoringsToProcess.isEmpty()) {
-					allRefactoringCommits = refactoringAnalyzer.collectCommitData(currentCommit, superCommitMetaData, refactoringsToProcess, entries);
+					allRefactoringCommits = refactoringAnalyzer.collectCommitData(currentCommit, superCommitMetaData,
+							refactoringsToProcess, entries);
 				} else if (thereIsRefactoringToProcess) {
+					String errorState = LogUtils.createErrorState(commitHash, project);
 					// timeout happened, so count it as an exception
-					log.debug("Refactoring Miner did not find any refactorings for commit: " + commitHash + createErrorState(commitHash, project));
+					log.debug("Refactoring Miner did not find any refactorings for commit: {} {}", commitHash,
+							errorState);
 				} else {
+					String errorState = createErrorState(commitHash, project);
 					// timeout happened, so count it as an exception
-					log.error("Refactoring Miner timed out for commit: " + commitHash + createErrorState(commitHash, project));
+					log.error("Refactoring Miner timed out for commit: {} {}", commitHash, errorState);
 					exceptionsCount++;
 				}
 			}
 
-			//collect the process metrics for the current commit
+			// collect the process metrics for the current commit
 			Set<ImmutablePair<String, String>> refactoringRenames = getRefactoringMinerRenames(refactoringsToProcess);
 			Set<ImmutablePair<String, String>> jGitRenames = getJGitRenames(entries);
-			processMetrics.collectMetrics(currentCommit, superCommitMetaData, allRefactoringCommits, entries, refactoringRenames, jGitRenames);
+			processMetrics.collectMetrics(currentCommit, superCommitMetaData, allRefactoringCommits, entries,
+					refactoringRenames, jGitRenames);
 			long startTimeTransaction = System.currentTimeMillis();
-			db.commit();
-			log.debug("Committing the transaction for commit " + commitHash + " took " + (System.currentTimeMillis() - startTimeTransaction) + " milliseconds.");
+			log.debug("Committing the transaction for commit {} took {} milliseconds.", commitHash,
+					(System.currentTimeMillis() - startTimeTransaction));
 		} catch (Exception e) {
+			// are not supported by mariadb
 			exceptionsCount++;
-			log.error("Unhandled exception when collecting commit data for commit: " + commitHash + createErrorState(commitHash, project), e);
-			db.rollback(createErrorState(commitHash, project));
```

Comment:
Why did you remove the transaction rollback?
In my eyes it is a reliable and save way to remove the last commit from the DB. Which might not even be the best idea at all, as we lose some data points, that we could keep and clear up the DB later.

Reply:
I wanted to encapsulate a single run for a project in a single transaction. So that if an error outside the commit analysis occurs there is no data related to that project is left over in the db. Unfortunately most open source db engines don't support nested transactions as far as i know. I came up with a solution that keeps track of the persisted entities during the commit analysis and removes them in the session if any errors occur. However in hindsight it seems that ll persists are of the commits with the lone exception of saving the project. So the previous solution suffices.